From bhutchings@solarflare.com  Thu Jun 23 20:54:30 2011
From: Ben Hutchings <bhutchings@solarflare.com>
Date: Mon, 23 Nov 2009 16:05:28 +0000
Subject: [PATCH 042/266] sfc: Move definition of struct falcon_nic_data into
 falcon.h
Patch-mainline: 2.6.39
Git-commit: 5c16a96c4f31a0be287c5db3f36d1099dea9b2bd
References: bnc#698572, FATE#311724

This is preparation for moving Falcon-specific state required by other
Falcon-specific code.

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/sfc/falcon.c |   11 -----------
 drivers/net/sfc/falcon.h |   11 +++++++++++
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/drivers/net/sfc/falcon.c b/drivers/net/sfc/falcon.c
index 29e79f7..490bda0 100644
--- a/drivers/net/sfc/falcon.c
+++ b/drivers/net/sfc/falcon.c
@@ -14,7 +14,6 @@
 #include <linux/module.h>
 #include <linux/seq_file.h>
 #include <linux/i2c.h>
-#include <linux/i2c-algo-bit.h>
 #include <linux/mii.h>
 #include "net_driver.h"
 #include "bitfield.h"
@@ -33,16 +32,6 @@
  * present in SFE400X evaluation boards
  */
 
-/**
- * struct falcon_nic_data - Falcon NIC state
- * @pci_dev2: The secondary PCI device if present
- * @i2c_data: Operations and state for I2C bit-bashing algorithm
- */
-struct falcon_nic_data {
-	struct pci_dev *pci_dev2;
-	struct i2c_algo_bit_data i2c_data;
-};
-
 /**************************************************************************
  *
  * Configurable values
diff --git a/drivers/net/sfc/falcon.h b/drivers/net/sfc/falcon.h
index 54dad2d..46dd43b 100644
--- a/drivers/net/sfc/falcon.h
+++ b/drivers/net/sfc/falcon.h
@@ -11,6 +11,7 @@
 #ifndef EFX_FALCON_H
 #define EFX_FALCON_H
 
+#include <linux/i2c-algo-bit.h>
 #include "net_driver.h"
 #include "efx.h"
 
@@ -29,6 +30,16 @@ static inline int falcon_rev(struct efx_nic *efx)
 	return efx->pci_dev->revision;
 }
 
+/**
+ * struct falcon_nic_data - Falcon NIC state
+ * @pci_dev2: The secondary PCI device if present
+ * @i2c_data: Operations and state for I2C bit-bashing algorithm
+ */
+struct falcon_nic_data {
+	struct pci_dev *pci_dev2;
+	struct i2c_algo_bit_data i2c_data;
+};
+
 static inline struct falcon_board *falcon_board(struct efx_nic *efx)
 {
 	return &efx->board_info;
-- 
1.7.4.4

